### DISCUSSION TITLE
How to change instance type for existing cluster?

### DISCUSSION THREAD

**Discussion by allantatter** [vitobotta/hetzner-k3s#455] [Q&A] [answered]
I had a cluster created during v1 era. I now upgraded my cluster.yaml according to the v2 upgrade guide (https://github.com/vitobotta/hetzner-k3s/releases/tag/v2.0.0). Upgrading was successful, re-running `hetzner-k3s create` had successful results.

My cluster has a single master node with `schedule_workloads_on_masters: true`. My question is, how to change the instance type of the master node? I want to increase the resources, from ccx13 to ccx23. When changing it from the yaml and re-running `hetzner-k3s create` then I ended up with a new cluster (https://d.pr/i/6BRdlD) which is probably not what I wanted to end up with. Should I leave the yaml with ccx13 config and change the instance type from Hetzner UI? Or what would be the preferred scenario in my situation? Thanks in advance.

<img width="389" alt="Markup 2024-09-24 at 09 39 44" src="https://github.com/user-attachments/assets/6e5db5b8-4b58-4ca8-a763-526db4c01394">

**Comment by vitobotta** [comment, marked as answer]
You cannot change the instance type from the config file. hetzner-k3s won't automatically change the instance type in that case and there is a risk of ruining the cluster by doing that. Is your cluster still operational?

The correct way is to just rescale the server from the Hetzner UI and leave the instance type untouched in the config file. Unfortunately adding the instance type to the instance name was an oversight in v1 since I hadn't thought about instance type changes. With v2, by default the instance type is no longer added to the instance name, so you can just rescale instances from the UI without any confusion.

**Comment by allantatter** [comment]
@vitobotta Thanks for a quick answer. My original cluster still works, deleted the new one. Then I know, that instances should be scaled from the UI, not from the config. Is it the same with clusters created with v2? No worries about the API changes v1->v2, that's what major upgrades are for :)

**Reply by vitobotta** [reply]
Yep it's the same with v2. The advantage is that with v2 there isn't confusion due to the mismatch of the instance type in the instance name :)
